junit-jupiter-api/src/main/java/org/junit/jupiter/api/DisplayNameGenerator.java

```diff
@@ -141,4 +168,131 @@ private static boolean hasParameters(Method method) {
 		}
 	}
 
+	/**
+	 * {@code DisplayNameGenerator} that generates complete sentences.
+	 *
+	 * <p>This implements the functionality of {@link DisplayNameGenerator}
+	 * by generating complete sentences display names, these
+	 * sentences are divided with a separator, and the generator and separator
+	 * can be customisable by using the {@link IndicativeSentencesGeneration}
+	 * interface as annotation.
+	 *
+	 * @since 5.7
+	 */
+	@API(status = EXPERIMENTAL, since = "5.7")
+	class IndicativeSentences implements DisplayNameGenerator {
+
+		static final DisplayNameGenerator INSTANCE = new IndicativeSentences();
+
+		@Override
+		public String generateDisplayNameForClass(Class<?> testClass) {
+			return getGeneratorForIndicativeSentence(testClass).generateDisplayNameForClass(testClass);
+		}
+
+		@Override
+		public String generateDisplayNameForNestedClass(Class<?> nestedClass) {
+			return classReplaceToIndicativeSentence(nestedClass);
+		}
+
+		@Override
+		public String generateDisplayNameForMethod(Class<?> testClass, Method testMethod) {
+			return classReplaceToIndicativeSentence(testClass) + getSentenceSeparator(testClass)
+					+ getGeneratorForIndicativeSentence(testClass).generateDisplayNameForMethod(testClass, testMethod);
+		}
+
+		private String classReplaceToIndicativeSentence(Class<?> testClass) {
+			Class<?> enclosingParent = testClass.getEnclosingClass();
+			Optional<DisplayName> displayName = AnnotationSupport.findAnnotation(testClass, DisplayName.class);
+			Optional<DisplayNameGeneration> displayNameGeneration = AnnotationSupport.findAnnotation(testClass,
+				DisplayNameGeneration.class);
+
+			if (enclosingParent == null) {
+				return displayName.map(DisplayName::value).orElseGet(() -> generateDisplayNameForClass(testClass));
+			}
+			else {
```

Comment:
I don't know if I already mentioned that, but personally I would get rid of both `else` blocks, since a value has already been returned in the `if` block. I feel that in most cases it makes the code more readable. What do you think?

@junit-team/junit-lambda Is there some kind of tacit style rule about this? 😛

Reply:
Sure I didn't think about it, I already make this changes on mi branch and look indeed more readable, thanks this changes comes in the next commit.